### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by Luni-4**
- Refactor Linux CI
- Add Windows CI based on Mingw64
- Add a tutorial to README to build examples through `vcpkg`

### CODE REVIEW

**Comment by Luni-4** [comment]
@asuper0

I don't know why but Windows CI doesn't work, I've installed `pkg-config` and it works and the two packages. I set up the environment variable but without any result. Do you know what could be the problem?

**Comment by asuper0** [comment]
vcpkg build packages from source code, so it will take a lot of time rather than apt in ubuntu. By default, vcpkg build both debug and release version. You can disable debug version by add `set(VCPKG_BUILD_TYPE release)` to triplets/x64-windows.cmake , that would save time.

**Comment by Luni-4** [comment]
> vcpkg build packages from source code, so it will take a lot of time rather than apt in ubuntu. By default, vcpkg build both debug and release version. You can disable debug version by add `set(VCPKG_BUILD_TYPE release)` to triplets/x64-windows.cmake , that would save time.

Thank you for the advice! I'm going to add that later, but for now the problem is that I can't build the code. It seems there is an hindrance [here](https://github.com/rust-av/examples/runs/7895210742?check_suite_focus=true#step:7:240)

**Comment by asuper0** [comment]
I noticed [CI log](https://github.com/rust-av/examples/runs/7895210742?check_suite_focus=true#step:7:241). There is a bug in vcpkg that the version info in opus.pc is "0". You may modify yourself, it's 1.3.1, I guess.

**Comment by Luni-4** [comment]
> I noticed [CI log](https://github.com/rust-av/examples/runs/7895210742?check_suite_focus=true#step:7:241). There is a bug in vcpkg that the version info in opus.pc is "0". You may modify yourself, it's 1.3.1, I guess.

How can I do that?

**Comment by asuper0** [comment]
the file vcpkg\installed\x64-windows\lib\pkgconfig\opus.pc . It looks like
```
prefix=${pcfiledir}/../..
# Opus codec reference implementation pkg-config file

exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include

Name: Opus
Description: Opus IETF audio codec ( build)
URL: https://opus-codec.org/
Version: 0
Requires:
Conflicts:
Libs: -L"${prefix}/lib" -lopus
Libs.private: 
Cflags: -I"${prefix}/include/opus"

```

Change `Version: 0` to `Version: 1.3.1` or `Version: 1.3.1-34-g5c94ec32`

Here is the issue and they fixed it but not merge yet.
[Issue](https://github.com/microsoft/vcpkg/issues/26369)

**Comment by Luni-4** [comment]
@asuper0 
Yeah, it works now! Unfortunately, I wasn't able to shatter down `libvpx` building time even though it seems we are in `Release mode`

**Comment by asuper0** [comment]
I find out that is because the [config in libvpx](https://github.com/microsoft/vcpkg/blob/master/ports/libvpx/portfile.cmake#L275-L279). It is set to not build debug by default.
BTW, the [PR in vcpkg](https://github.com/microsoft/vcpkg/pull/26389) is merged today.

**Comment by Luni-4** [comment]
> I find out that is because the [config in libvpx](https://github.com/microsoft/vcpkg/blob/master/ports/libvpx/portfile.cmake#L275-L279). It is set to not build debug by default. 

So can I remove the `set(VCPKG_BUILD_TYPE release)` part?

>BTW, the [PR in vcpkg](https://github.com/microsoft/vcpkg/pull/26389) is merged today.

Perfect, let me see if I can remove the hack step from CI

**Comment by asuper0** [comment]
> So can I remove the set(VCPKG_BUILD_TYPE release) part?

No, that can save time while building opus and other packages in the future. I advise to keep it.

File: README.md
Comment (asuper0) [inline README.md]: I suggest to mark this as optional

File: README.md
Comment (asuper0) [inline README.md]: append "windows x64" in the label

Comment (Luni-4) [reply README.md]: Which label?

File: README.md
Comment (Luni-4) [reply README.md]: Yes, you're right

**Comment by Luni-4** [comment]
@asuper0 

Thank you for your review! I should have solved and clarified everything, as soon as you approve this PR I'm going to merge it.

**Review by asuper0 — APPROVED** [review]
That's good!

**Pull request merged**
